cfme/utils/providers.py

```diff
@@ -358,6 +358,7 @@ def get_mgmt(provider_key, providers=None, credentials=None):
     # TODO rename the parameter; might break things
     if isinstance(provider_key, Mapping):
         provider_data = provider_key
+        provider_key = provider_key['type']
```

Comment:
I don't think that this is an assumption that we can make in all cases when a dictionary is passed instead of a bare key.

There are checks throughout the rest of the method that check provider_key type before acting on it, and I think we should consider modifying the check on L389/390 that's raising the exception to make the check based on the `provider_key` type.  See below for example.